chalice/awsclient.py

```diff
@@ -172,25 +172,12 @@ def _create_lambda_function(self, api_args):
             )
             raise self._get_lambda_code_deployment_error(e, context)
 
-    def _call_client_method_with_retries(self, method, kwargs):
-        # type: (_CLIENT_METHOD, Dict[str, Any]) -> Dict[str, Any]
-        client = self._client('lambda')
-        attempts = 0
-        while True:
-            try:
-                response = method(**kwargs)
-            except client.exceptions.InvalidParameterValueException as e:
-                # We're assuming that if we receive an
-                # InvalidParameterValueException, it's because
-                # the role we just created can't be used by
-                # Lambda so retry until it can be.
-                self._sleep(self.DELAY_TIME)
-                attempts += 1
-                if attempts >= self.LAMBDA_CREATE_ATTEMPTS or \
-                        not self._is_iam_role_related_error(e):
-                    raise
-                continue
-            return response
+    def _is_settling_error(self, error):
+        # type: (botocore.exceptions.ClientError) -> bool
+        message = error.response['Error'].get('Message', '')
+        if re.search('event source mapping.*is in use', message):
```

Comment:
Do you think it would be better to check the `Code`?
```python
response['Error'].get('Code', '') == 'ResourceInUseException'
```

Reply:
Yeah I would like to know the reasoning of going with message approach as well.

Reply:
This is handled by catching the `ResourceInUseException` here: https://github.com/aws/chalice/pull/886/files/0096cd0ce45857ad2cfad23e8138c50055ca17a0#diff-3aa85da792f26ee5a92aff6a3d060cccR957.

If we knew that this error was only raised when the event source mapping is still settling, then we could remove this check, but I'm not sure if that's a safe assumption.

Reply:
I say we keep the message check in place as well. I'm not sure if it is a safe assumption to make either.

Reply:
Make sense.